app/controllers/posts_controller.rb

```diff
@@ -1,14 +1,20 @@
 class PostsController < ApplicationController
+  before_action :authenticate_user!
+
```

Comment:
- [ ]  You did great job to configure Cancan to implement the authorization in you project.  and I like how you are using `authorize!` method to check for the authority on destroy method in both posts and comment controller actions.  One thing I would suggest you to use `load_and_authorize_resource `method to handle the loading and authorize other actions in your post controller.
`load_and_authorize_resource `is a convenient way to automatically load and authorize resources for multiple actions in a controller

Reply:
Yes convenience... Thank @ishimwezachee